### Debugging 

In [1]:
import pandas as pd 
import numpy as np


In [2]:
env_df= pd.read_csv('data/raw/Composite.csv')
PID_df=pd.read_csv('data/lookup/PID_location_all.csv')
traits=pd.read_csv("data/lookup/traitMatrix.csv")


PID_df.drop_duplicates(subset=['PID'], inplace=True)

In [7]:
PID_df

,PID,lat,lon,BHAGE,managed,ownership,biome
0,1_32_3_827_420,35.233708,-114.742600,NaN,0.0,blm,Xeric shrublands
3,1_32_3_822_420,35.266943,-114.736182,NaN,0.0,blm,Xeric shrublands
8,1_32_3_85923_1,35.268947,-114.744102,NaN,0.0,blm,Xeric shrublands
12,1_32_3_760_420,35.549378,-115.178318,NaN,0.0,blm,Xeric shrublands
21,1_32_3_82579_1,35.551445,-115.178240,NaN,0.0,blm,Xeric shrublands
...,...,...,...,...,...,...,...
23500455,1_19_5_20090_1,43.490491,-91.400523,NaN,0.0,other,Temperate broadleaf forests
23500483,1_19_5_10220_1,43.491202,-91.229999,NaN,0.0,fish_wildlife,Temperate broadleaf forests
23500522,1_19_89_20059_1,43.491696,-92.156736,NaN,1.0,other,Temperate broadleaf forests
23500552,1_19_89_20059_323,43.491696,-92.156736,NaN,0.0,other,Temperate broadleaf forests


In [3]:
print(env_df.shape, PID_df.shape, traits.shape)

(313724, 48) (408201, 7) (54153, 19)


* Rescrape Composite Data

In [5]:
npp_df=pd.read_csv('data/raw/npp_10_years.csv')

In [ ]:
df1=pd.read_csv('data/processed/PID_npp.csv') # This is NPP merged with PID locations from 1b
df1.drop_duplicates(subset=['PID'], inplace=True)

print(df1.shape)

(124317, 4)


Merging with NPP data goes from  408 000 datapoints to 124 317

In [17]:
df = pd.read_parquet("data/final/dataset.parquet") # From 2-Main
df.drop_duplicates(subset=['PID'], inplace=True)

print(df.shape)


(399024, 25)


In [ ]:
fd_df=pd.read_csv('data/final/fd_df.csv')
print(fd_df.shape)


dataset.parquet joins FIA state data with Composite data (313724) and PID location (408201) along with the diversity metrics (calculated from trait matrix)
* How did I get from 313 composite data rows to 399 000 rows??????????
